packages/common/src/ContractInteractor.ts

```
@@ -574,11 +578,38 @@ export class ContractInteractor {
     return ret
   }
 
+  async _getPastEventsPaginated (
+    contract: any,
+    names: EventName[],
+    extraTopics: string[],
+    options: PastEventOptions
+  ): Promise<EventData[]> {
+    const MAX_RETRY_COUNT = 10
+    const INITIAL_DELAY = 1000
+    const MAXIMUM_DELAY = 500000
+    const STEP = 2
+    let delay = INITIAL_DELAY
+    for (let i = 0; i < MAX_RETRY_COUNT; i++) {
+      try {
+        return await this._getPastEventsPaginatedInternal(contract, names, extraTopics, options)
+      } catch (error) {
+        console.error('_getPastEventsPaginated failed', error)
+        if (!this.getPastEventsExponentialBackoff) {
+          throw error
+        }
+        // exponential backoff up to 1000 sec = 16 minutes
+        await this.sleep(delay)
+        delay = Math.min(delay * STEP, MAXIMUM_DELAY)
+      }
+    }
+    throw new Error(`_getPastEventsPaginated failed after ${MAX_RETRY_COUNT} attempts`)
```

you lost the error here. do it inside the loop, so you can throw the nth error